frontend/src/core/codemirror/language/__tests__/utils.test.ts

```diff
@@ -147,3 +148,54 @@ describe("splitEditor", () => {
     expect(result.afterCursorCode).toEqual('mo.md(f"""{b}!""")');
   });
 });
+
+describe("extractHighlightedCode", () => {
+  const firstLine = "import marimo as mo";
+  const secondLine = "import pandas as pd";
+  const thirdLine = "import numpy as np";
+
+  it("extracts highlighted code at start", () => {
+    const mockEditor = createEditor(
+      `${firstLine}\n${secondLine}\n${thirdLine}`,
+    );
+    mockEditor.dispatch({
+      selection: {
+        anchor: mockEditor.state.doc.line(1).from,
+        head: mockEditor.state.doc.line(2).to,
+      },
+    });
+    const [highlighted, leftover] = extractHighlightedCode(mockEditor);
+    expect(highlighted).toEqual(`${firstLine}\n${secondLine}`);
+    expect(leftover).toEqual(`\n${thirdLine}`);
+  });
+
+  it("extracts highlighted code in middle", () => {
+    const mockEditor = createEditor(
+      `${firstLine}\n${secondLine}\n${thirdLine}`,
+    );
+    mockEditor.dispatch({
+      selection: {
+        anchor: mockEditor.state.doc.line(2).from,
+        head: mockEditor.state.doc.line(2).to,
+      },
+    });
+    const [highlighted, leftover] = extractHighlightedCode(mockEditor);
+    expect(highlighted).toEqual(secondLine);
+    expect(leftover).toEqual(`${firstLine}\n${thirdLine}`);
+  });
+
+  it("extracts highlighted code at end", () => {
+    const mockEditor = createEditor(
+      `${firstLine}\n${secondLine}\n${thirdLine}`,
+    );
+    mockEditor.dispatch({
+      selection: {
+        anchor: mockEditor.state.doc.line(3).from,
+        head: mockEditor.state.doc.line(3).to,
+      },
+    });
+    const [highlighted, leftover] = extractHighlightedCode(mockEditor);
+    expect(highlighted).toEqual(thirdLine);
+    expect(leftover).toEqual(`${firstLine}\n${secondLine}`);
+  });
+});
```

Comment:
can you write some tests for markdown and sql?
1. sql x no selection
2. sql x with selection
3. markdown x no selection
4. markdown x with selection